src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -563,6 +578,9 @@ public AgentSettings LoadSettings()
                     // delete agent cert setting
                     (HostContext.GetService<IAgentCertificateManager>() as AgentCertificateManager).DeleteCertificateSetting();
 
+                    // delete agent runtime option
+                    _store.DeleteAgentRuntimeOptions();
```

Comment:
should this be best effort? is it different than deleting the rest, or should it just be consistent?